sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -32,39 +32,37 @@ stripe_customers AS (
   FROM
     mozdata.stripe.customers
 ),
-fxa_logs_provider_country AS (
+paypal_country AS (
   SELECT
     fxa_uid,
     plan_id,
-    ARRAY_AGG(
-      STRUCT(payment_provider AS provider, source_country AS country)
-      ORDER BY
-        event_timestamp DESC
-      LIMIT
-        1
-    )[OFFSET(0)].*,
+    ARRAY_AGG(source_country ORDER BY event_timestamp DESC LIMIT 1)[OFFSET(0)] AS country,
   FROM
     `moz-fx-data-shared-prod`.stripe_external.fxa_pay_setup_complete_v1
+  WHERE
+    LOWER(payment_provider) = "paypal"
   GROUP BY
     fxa_uid,
     plan_id
 ),
+stripe_charges AS (
+  SELECT
+    id AS charge_id,
+    payment_method_details.card.country,
+  FROM
+    `moz-fx-data-shared-prod`.stripe_external.charges_v1
+  WHERE
+    status = "succeeded"
+),
 stripe_invoice_lines AS (
   SELECT
     lines.subscription AS subscription_id,
-    INITCAP(
-      COALESCE(
-        fxa_logs_provider_country.provider,
-        IF(
-          "paypalTransactionId" IN (SELECT key FROM UNNEST(invoices_v1.metadata)),
-          "Paypal",
-          "Stripe"
-        )
-      )
-    ) AS provider,
-    LOWER(
-      COALESCE(fxa_logs_provider_country.country, charges_v1.payment_method_details.card.country)
-    ) AS country,
+    IF(
+      "paypalTransactionId" IN (SELECT key FROM UNNEST(invoices_v1.metadata))
+      OR (invoices_v1.charge IS NULL AND invoices_v1.status = "paid"),
```

Comment:
see https://bugzilla.mozilla.org/show_bug.cgi?id=1712027#c1 for why presence of `"paypalTransactionId"` alone is insufficient

Reply:
Should the reference to the bug be documented directly into the query?